src/middlewared/middlewared/plugins/kerberos.py

```diff
@@ -166,6 +172,24 @@ async def do_update(self, data):
         await super().do_update(data)
 
         await self.middleware.call('etc.generate', 'kerberos')
+
+        # gssproxy provides support for large kerberos tickets.
+        # It does not replace the kernel gssapi. The kernel will use gssproxy
+        # if it is available.  This is determined during the boot process.
+        # Currently, the status of gssproxy cannot be changed outside of boot time.
+        if new['enable_gssproxy'] != old['enable_gssproxy']:
+            # gssproxy seems to ignore the 'disable' setting and requires masking
+            if new['enable_gssproxy'] is True:
+                await systemd_unit("gssproxy", "unmask")
+                await systemd_unit("gssproxy", "enable")
+                # Reboot is required to enact the change
+                self.logger.info("Processed gssproxy enable. Reboot is required to enact change.")
+            else:
+                await systemd_unit("gssproxy", "disable")
+                await systemd_unit("gssproxy", "mask")
+                # Reboot is required to enact the change
+                self.logger.info("Processed gssproxy disable. Reboot is required to enact change.")
```

Comment:
I think we should also add a "reboot required" FilePresenceAlert so that it gets visibility with end-user. c.f. UpdateFailedAlertSource

You'll probably want to put the sentinel file in the middleware rundir (tmpfs) so that it gets auto-cleared on reboot.

Reply:
Or make it a one-shot alert that is removed on boot, more modern approach 🙂

Reply:
Okay. I pointed Mark at some examples of one-shot alerts. @themylogin do you think it makes sense to create the alert on system update if we're not automatically rebooting?

Reply:
@anodos325 how is system update involved in this?

Reply:
Another situation where we might want to generate an alert "You should reboot your server" :)

Reply:
Removed enable/disable capability